model/shared/src/main/scala/queries/schemas/itc/syntax.scala

```diff
@@ -5,31 +5,65 @@ package queries.schemas.itc
 
 import cats.Hash
 import cats.syntax.all.*
+import eu.timepit.refined.types.numeric.PosDouble
 import explore.model.AsterismIds
 import explore.model.TargetList
 import explore.model.itc.ItcTarget
 import explore.modes.GmosNorthSpectroscopyRow
 import explore.modes.GmosSouthSpectroscopyRow
 import explore.modes.InstrumentRow
 import explore.optics.all.*
+import lucuma.core.enums.GmosNorthFpu
+import lucuma.core.enums.GmosSouthFpu
+import lucuma.core.enums.GmosXBinning
+import lucuma.core.enums.GmosYBinning
+import lucuma.core.enums.ImageQuality
 import lucuma.core.math.RadialVelocity
 import lucuma.core.model.*
+import lucuma.core.model.sequence.gmos.GmosCcdMode
+import lucuma.core.model.sequence.gmos.longslit.*
 import lucuma.itc.client.GmosFpu
 import lucuma.itc.client.InstrumentMode
 
 trait syntax:
 
   extension (row: InstrumentRow)
-    def toItcClientMode: Option[InstrumentMode] = row match {
+    def toItcClientMode(p: SourceProfile, iq: ImageQuality): Option[InstrumentMode] = row match {
       case g: GmosNorthSpectroscopyRow =>
-        val roi = g.modeOverrides.flatMap(_.roi)
-        val ccd = g.modeOverrides.flatMap(_.ccdMode)
+        val isFPU =
```

Comment:
Is this the same logic as in the ODB? If so, can we move it to a common lib? core?

Reply:
Yes I added this as a method to core

Reply:
Nice, I was wondering if the could unify the override logic, but I see now that the overrides are defined in explore.